src/SwcCompiler.ts

```diff
@@ -0,0 +1,204 @@
+import { Output, transformFile } from "@swc/core";
+import findRoot from "find-root";
+import globby from "globby";
+import { Compiler } from "./Compiler";
+import { ProjectConfig } from "./Options";
+import { log, projectConfig } from "./utils";
+
+// https://esbuild.github.io/api/#resolve-extensions
+const DefaultExtensions = [".tsx", ".ts", ".jsx", ".mjs", ".cjs", ".js"];
+
+export type CompiledFile = Output & { filename: string; root: string };
+export type Group = { root: string; files: Array<CompiledFile> };
+
+type FileGroup = Map<string, CompiledFile>;
+class CompiledFiles {
+  private groups: Map<string, FileGroup>;
+
+  constructor() {
+    this.groups = new Map();
+  }
+
+  addFile(file: CompiledFile) {
+    let group = this.groups.get(file.root);
+    if (!group) {
+      group = new Map();
+      this.groups.set(file.root, group);
+    }
+    group.set(file.filename, file);
+  }
+
+  group(filename: string): Group | undefined {
+    for (const [root, files] of this.groups.entries()) {
+      if (files.get(filename)) {
+        return { root, files: Array.from(files.values()) };
+      }
+    }
+  }
+}
+
+/** Implements TypeScript building using esbuild */
+export class SwcCompiler implements Compiler {
+  private compiledFiles: CompiledFiles;
+  constructor(readonly workspaceRoot: string) {
+    this.compiledFiles = new CompiledFiles();
+  }
+
+  async invalidateBuildSet() {
+    this.compiledFiles = new CompiledFiles();
+  }
+
+  async compile(filename: string) {
+    const existingGroup = this.compiledFiles.group(filename);
+
+    if (existingGroup) {
+      await this.buildFile(filename, existingGroup.root);
+    } else {
+      await this.buildGroup(filename);
+    }
+
+    return (this.compiledFiles.group(filename)! as unknown) as void;
+  }
+
+  async fileGroup(filename: string) {
+    const contents: Record<string, string> = {};
+    const group = this.compiledFiles.group(filename);
+
+    if (!group) {
+      throw new Error(await this.missingDestination(filename));
+    }
+
+    for (const file of group.files) {
+      contents[file.filename] = file.code;
+    }
+
+    return contents;
+  }
+
+  private async getModule(filename: string) {
```

Comment:
A few methods in this file have been copied verbatim from the `EsBuildCompiler`. If we keep both compilers around it would be a good idea to deduplicate them.

Reply:
I agree -- I will ponder what I think the long term vision for the tool should be and if we really want to support two backends forever. It doesn't seem like that huge a maintenance burden and they do support strange and different feature sets.